src/mlpack/core/data/string_encoding_policies/tf_idf_encoding_policy.hpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * @file td_idf_encoding_policy.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of the TfIdfEncodingPolicy class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_ENCODING_POLICIES_TF_IDF_ENCODING_POLICY_HPP
+#define MLPACK_CORE_DATA_ENCODING_POLICIES_TF_IDF_ENCODING_POLICY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/data/string_encoding_policies/policy_traits.hpp>
+#include <mlpack/core/data/string_encoding.hpp>
+namespace mlpack {
+namespace data {
+/**
+ * Definition of the TfIdfEncodingPolicy class.
+ *
+ * Tf-idf is weighing scheme that stands for term-frequency multiplied by
+ * inverse document-frequency.
+ * The goal of using tf-idf is to scale down the impact of tokens that occur
+ * very frequently in a given corpus while using the type of Term-Frequency
+ * of a token and that are hence empirically less informative than features
+ * that occur in a small fraction of the training corpus.
+ * TfIdfEncodingPolicy is used as a helper class for StringEncoding.
+ * The encoder assigns a tf-idf number to each unique token and treat 
+ * the dataset as categorical. The tokens are labeled in the order of their
+ * occurrence in the input dataset.
+ */
+class TfIdfEncodingPolicy
+{
+ public:
+  /**
+  * Enum class used to identify the type of tf encoding
+  *
+  * Follwing are the defination of the types
+  * BINARY : binary weighting scheme (0,1)
+  * RAW_COUNT : raw count weighting scheme (count of token for every row)
+  * TERM_FREQUENCY : term frequency weighting scheme (count / length(row))
+  * SUBLINEAR_TF : logarthimic weighting scheme (log(tf) + 1) 
+  * 
+  */
+  enum class TfTypes
+  {
+    RAW_COUNT,
+    BINARY,
+    SUBLINEAR_TF,
+    TERM_FREQUENCY,
+  };
+
+  /**
+  * A constructor for the class which is use to set the type of term frequency
+  * and also the value for somoothIdf.
+  *
+  * @param tfType The type of term frequency, The avialbale option are
+  *     RAW_COUNT : The count of a specific token
+  *     BINARY : 1 if token occurs in document and 0 otherwise;
+  *     TERM_FREQUENCY :  Raw_count ÷ (number of words in document)
+  *     SUBLINEAR_TF : log(Raw_Count) + 1
+  *
+  * @param smoothIdf Used to indicate whether to use smooth idf or not.
+  */
```

Comment:
```suggestion
   * A constructor for the class which is use to set the type of term frequency
   * and also the value for somoothIdf.
   *
   * @param tfType The type of term frequency, The available options are
   *     RAW_COUNT : The count of a specific token
   *     BINARY : 1 if token occurs in document and 0 otherwise;
   *     TERM_FREQUENCY :  Raw_count ÷ (number of words in document)
   *     SUBLINEAR_TF : log(Raw_Count) + 1
   *
   * @param smoothIdf Used to indicate whether to use smooth idf or not.
   */
```